### PR TITLE
Added new Feeder settings panel, added support for multiple feeder outputs with differing formats, added new JSON format -- JSONdump

### LINKED ISSUE #54: Resolve hostname in UDP stream.
In Settings -> ACARS window I would like to be able to add a hostname and port for data to be sent to.    
At the moment only lP:port function.     
Many people are running Jaero and feeding ACARS data to a central website. If the IP address changes all Jaeros need to be updated. Enabling Jaero to resolve the hostname would solve this issue.    
     
Thanks for your work.

### PR DESCRIPTION
**Pull request by andermatt64**
## Note
Apologies for the gigantic PR, my original PR was smaller but after talking to others, there's a real need for multiple feeder outputs with differing formats.

## What changed?
* Updated Windows CI build script to support latest versions
* New settings panel **Feeder** that gives users an option to define multiple feed output servers (and their desired formats)
* Refactored message formatting to its own, mostly idempotent, function (at least within `ACARSslot`)
* Refactored uses of `QString::sprintf` to use `QString::arg` -- manually verified that output does not differ with said changes for formats 1,2,3,JSON,JSONdump with the provided sample
* Ensured backwards compat with older JAERO settings files. Old feed host/port servers are automatically added as feed outputs.
* Feed server target can now be host names -- no need to use only IP addresses anymore. (closes #54)
* Fixed typo with `populatepublicvars`
* Added generated files `JAERO` and `moc_predefs.h` to git ignore
* Added option to only use libacars library to parse downlink ADS messages
* Added option to output JSON format (JSONdump) similar to @szpajder 's dumphfdl and dumpvdl2 
* libacars decoded messages will exposed as JSON objects if in JSON4 format
* Added settings options for inputting station ID (default is hostname)

## Testing
* Verified refactored `QString::arg` does not alter output formats with provided sample WAVs
* Manually tested most common situations with feeder UI flow -- (could use more work here)
* Manually tested some simple feeding situations such as 3 feeder inputs to different ports on the same host -- (could use more testing here)
* Very casually took a look a memory usage, doesn't _appear_ to be leaking.

### Settings Dialog Changes
Newly added options to Settings -> Decoding panel
image

Newly added Settings -> Feeder panel
image

Newly added create/edit dialog for feeders
image

## Example JSONdump Format Output
<pre>
{
  "app": {
    "name": "JAERO",
    "ver": "v1.0.4.13"
  },
  "isu": {
    "acars": {
      "ack": "!",
      "arinc622": {
        "adsc": {
          "err": false,
          "tags": [
            {
              "wpt_change_event": {
                "alt": 36000,
                "lat": 51.001282,
                "lon": -30.005379,
                "nav_redundancy": true,
                "pos_accuracy_nm": 0.25,
                "tcas_avail": true,
                "ts_sec": 742
              }
            },
            {
              "predicted_route": {
                "next_next_wpt": {
                  "alt": 36000,
                  "lat": 50.999908,
                  "lon": -49.999981
                },
                "next_wpt": {
                  "alt": 36000,
                  "eta_sec": 3026,
                  "lat": 50.999908,
                  "lon": -40.000019
                }
              }
            }
          ]
        },
        "air_addr": ".D-AIHV",
        "crc_ok": true,
        "gs_addr": "PIKCPYA",
        "msg_type": "adsc_msg"
      },
      "blk_id": "7",
      "flight": "LH424",
      "label": "B6",
      "mode": "2",
      "msg_text": "J30ALH0424/PIKCPYA.ADS.D-AIHV14244487554D88CA0B981D0D2444471C71C8CA0BD2244446E38E48CA0097F3",
      "reg": "D-AIHV"
    },
    "dst": {
      "addr": "43",
      "type": "Ground Earth Station"
    },
    "qno": "07",
    "refno": "08",
    "src": {
      "addr": "3C6516",
      "type": "Aircraft Earth Station"
    }
  },
  "station": "jetson",
  "t": {
    "sec": 1665545421,
    "usec": 58000
  }
}
</pre>

### CODE REVIEW

File: JAERO/mainwindow.cpp [hunk 1]
```diff
@@ -1474,38 +1474,115 @@ void MainWindow::ACARSslot(ACARSItem &acarsitem)
...
-        if(!message.isEmpty())
+            //add database lookup info if available
+            if(acarsitem.dblookupresult.size()==QMetaEnum::fromType<DataBaseTextUser::DataBaseSchema>().keyCount())
+            {
+                json["DB_MANUFACTURER"]=acarsitem.dblookupresult[DataBaseTextUser::DataBaseSchema::Manufacturer].trimmed();
+                json["DB_TYPE"]=acarsitem.dblookupresult[DataBaseTextUser::DataBaseSchema::Type].trimmed();
+                json["DB_OWNERS"]=acarsitem.dblookupresult[DataBaseTextUser::DataBaseSchema::RegisteredOwners].trimmed();
+            }
+            //add common things
+            json["TIME"]=time.toSecsSinceEpoch();
```
Comment (kevinelliott) [inline JAERO/mainwindow.cpp:1488]: Any change of lowercasing this to match the others?

**Comment by kevinelliott** [comment]
Nice additions. I'd love to see this happen. Would be good to have the JSON keys match in case, but also to come even closer to @szpajder 's dump format (if possible) overall!

**Comment by kevinelliott** [comment]
Since we have the `app` element that indicates jaero and version, do you think the top level `jaero` element is necessary?

**Comment by andermatt64** [comment]
> Since we have the `app` element that indicates jaero and version, do you think the top level `jaero` element is necessary?

Practically speaking, I don't think it's necessary. I put it in just to closer match the dump JSON format

**Comment by kevinelliott** [comment]
@andermatt64 This is some really great work. Thanks for taking all of our feedback here and on the ADSBX Discord.

**Review by kevinelliott — APPROVED** [review]
Looks good to me! I think you could add `seqno` too.

**Comment by andermatt64** [comment]
> Looks good to me! I think you could add `seqno` too.

I tried adding it but the value seems to be always `00`. I suspect it may be a value used internally to reassemble fragmented messages

**Comment by kevinelliott** [comment]
Hey @jontio ... would love to see this merged. We have quite a few live streams using this to airframes.io by quite a few people. @andermatt64 has done a great job. Think you could take a look and consider it?

**Comment by thebaldgeek** [comment]
Can confirm. I have installed around 40 of these builds with zero issues.
Would really like to see it in the main build. Its tricky/confusing to point people to different repos.

**Comment by jontio** [comment]
Sorry I've been putting this off a bit as it's big and I've been doing other things recently. I was planning on going through it and do a bit of testing myself. But as people have tested it as seems to be working well I'll just pull it in. Thanks @andermatt64 and everyone else for their work.

**Pull request merged**